docs/user-guide/coordinate-transformations.ipynb

```diff
@@ -0,0 +1,359 @@
+   "source": [
+    "transformed = da.transform_coords(['x^2'], graph=graph)\n",
+    "transformed"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ab752aee-f2d7-4cae-8c7a-b33a2f7e8033",
+   "metadata": {},
+   "source": [
+    "Note how `x` is now an attribute, i.e., operations will not use it for alignment anymore.\n",
+    "This is important since it will allow for operations combining `transformated` with other data that may have matching `x^2` but not `x`.\n",
+    "\n",
+    "### Example: Multi-step transform splitting and combining input coords"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "833948c5-4657-43a7-bf38-46c385bd7e26",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import scipp as sc\n",
+    "import numpy as np\n",
+    "hour_steps = sc.arange(dim='timestamp', dtype='int64', unit='s', start=0, stop=3*24*60*60, step=60*60)\n",
+    "start = sc.scalar(np.datetime64('2021-01-01T17:00:00'))\n",
+    "datetime = start + hour_steps\n",
+    "\n",
+    "nsite = 1000\n",
+    "ntime = len(datetime)\n",
+    "# Note that these points are *note* uniformely distributed on a shpere, this is NOT a good way to\n",
+    "# generate such points\n",
+    "site = sc.vectors(dims=['cartesian'], values=np.random.rand(nsite, 3)) - sc.vector(value=[.5,.5,.5])\n",
+    "site /= sc.norm(site)\n",
+    "site *= 6.371 * sc.Unit('km')\n",
+    "site\n",
+    "da = sc.DataArray(\n",
+    "    data=sc.array(dims=['cartesian','timestamp'], values=np.random.rand(nsite, ntime)),\n",
+    "    coords={'cartesian':site, 'timestamp':datetime}\n",
+    ")\n",
+    "da += 2. * (site.fields.z > 0. * sc.Unit('km')).astype('float64')  # more lightning strikes in norther hemisphere\n",
+    "\n",
+    "phi0 = sc.atan2(y=site.fields.y, x=site.fields.x) - sc.to_unit(90.0 * sc.Unit('deg'), 'rad')\n",
+    "sin = sc.sin(phi0+sc.linspace(dim='timestamp', unit='rad', start=0, stop=6*np.pi, num=ntime))\n",
+    "(sin+1).plot()\n",
+    "da += 2*(sin+1)\n",
+    "da"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "757be4c3-8efc-4550-89e4-231cb53d0230",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "da.plot(projection='3d', positions=site)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b175bf2f-4543-4c5a-84cf-6e96d3a36f96",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def lat_long(cartesian):\n",
+    "    x = cartesian.fields.x\n",
+    "    y = cartesian.fields.y\n",
+    "    z = cartesian.fields.z\n",
+    "    theta = sc.atan2(y=sc.sqrt(x*x+y*y), x=z)\n",
+    "    phi = sc.atan2(y=y, x=x)\n",
+    "    return {'lattitude':theta, 'longitude':phi}\n",
+    "\n",
+    "def time(datetime):\n",
+    "    seconds_per_day = sc.scalar(24*60*60, unit='s/D')\n",
+    "    start_day = sc.scalar(start.value.astype('datetime64[D]'))\n",
+    "    start_day_in_seconds = sc.scalar(start_day.values.astype('datetime64[s]'))\n",
```

Comment:
Implies some missing `astype` conversion in scipp?

Reply:
Yes, because this is nontrivial.